src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceWrapper.cs

```diff
@@ -10,20 +12,27 @@ namespace Microsoft.AspNet.OData.Formatter.Deserialization
     /// Encapsulates an <see cref="ODataResource"/> and the inner nested resource infos.
     /// </summary>
     public sealed class ODataResourceWrapper : ODataItemBase
-    {
+    {       
         /// <summary>
         /// Initializes a new instance of <see cref="ODataResourceWrapper"/>.
         /// </summary>
         /// <param name="item">The wrapped item.</param>
-        public ODataResourceWrapper(ODataResource item)
+        public ODataResourceWrapper(ODataResourceBase item)
             : base(item)
         {
             NestedResourceInfos = new List<ODataNestedResourceInfoWrapper>();
+            ResourceBase = item;
         }
+  
+        /// <summary>
+        /// Gets the wrapped <see cref="ODataResourceBase"/>.
+        /// </summary>
+        public ODataResourceBase ResourceBase {get;}
 
         /// <summary>
-        /// Gets the wrapped <see cref="ODataResource"/>.
+        /// Gets the wrapped <see cref="ODataResource"/>. This will return null for deleted resources.
         /// </summary>
+        [Obsolete("Please use ResourceBase instead")]
```

Comment:
>[Obsolete("Please use ResourceBase instead")] [](http://example.com/codeflow?start=8&length=45)

This obsolete seems unnecessary.

Reply:
The idea was to make it obselete but not to remove, right @mikepizzo . but I guess in 8.0 its removed right?

Reply:
In 8.0 the Resource property should be typed as ODataResourceBase.